File: java/aula1/src/Ex6.java
import java.io.IOException;

public class Ex6 {

    public static void main(String[] args) {

        double smallest = 0, biggest = 0, current;

        System.out.println("Insira a sequencia a seguir: ");
        do {
            try {
                current = EntradaTeclado.leDouble();
                if (smallest == 0) {
                    smallest = current;
                }
                if (biggest == 0) {
                    biggest = current;
                }
                if (current != 0) {
                    smallest = Math.min(smallest, current);
                    biggest = Math.max(biggest, current);
                }

            } catch (IOException e) {
                System.out.println("Valor invalido inserido.");
                break;
            }
        } while (current != 0);

        System.out.printf("Maior: %f\nMenor: %f", biggest, smallest);
    }

}

File: python/oop2022/exs1/ex6.py
def read_float():
    while True:
        try:
            value = float(input())
            return value
        except ValueError:
            print("Valor invalido inserido, tente novamente:")


def main():
    print("Insira a sequencia a seguir:\n")

    biggest = 0
    smallest = 0

    current = 0

    while True:
        current = read_float()
        if current == 0:
            break

        if current < smallest or smallest == 0:
            smallest = current
        if current > biggest or biggest == 0:
            biggest = current

    print(f"Maior numero foi: {biggest}, e menor foi {smallest}")


if __name__ == "__main__":
    main()
